services/blockchain-indexer/shared/messageProcessor.js

```diff
@@ -77,8 +78,10 @@ const initQueueStatus = async () => {
 
 const newBlockProcessor = async header => {
 	logger.debug(`New block (${header.id}) received at height ${header.height}.`);
-	const response = await formatBlock(header);
+	const { data: transactions } = await getTransactionsByBlockID(header.id);
```

Comment:
Should we fetch `assets` as well? Since even `numberOfAssets` are inconsistent https://github.com/LiskHQ/lisk-service/issues/1930
If not maybe we can remove `numberOfAssets` from new block events?

Reply:
I'd also recommend to further optimize this. Do not make a call if the `transactionRoot` from the event payload matches the empty payload hash.

Let's discuss this offline whenever you're available.

Reply:
Applied.